hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieCDCLogger.java

```diff
@@ -275,7 +275,20 @@ private CDCTransformer getTransformer() {
   }
 
   private GenericRecord removeCommitMetadata(GenericRecord record) {
-    return record == null ? null : HoodieAvroUtils.rewriteRecordWithNewSchema(record, dataSchema.toAvroSchema(), Collections.emptyMap());
+    return record == null ? null : getRecordWithoutMetadata(record);
+  }
+
+  private GenericRecord getRecordWithoutMetadata(GenericRecord record) {
```

Comment:
yes, we can do this because there is prerequisite that no fields reordering or renaming, maybe we should check all the usages of `HoodieAvroUtils.rewriteRecordWithNewSchema` and replace it with this more performant way if it is the similiar use case.

Reply:
fire an issue to trace this: https://github.com/apache/hudi/issues/17679